rules/tick.go

```
@@ -101,32 +100,77 @@ func updateFood(width, height int64, gameFrame *pb.GameFrame, foodToRemove []*pb
 	return food, nil
 }
 
-func getUnoccupiedPoint(width, height int64, food []*pb.Point, snakes []*pb.Snake) (*pb.Point, error) {
-	attempts := 0
-	for {
-		attempts++
-		if attempts > 20 {
-			return nil, errors.New("unable to find available empty location after 20 attempts")
+func getUnoccupiedPoint(width int64, height int64, food []*pb.Point, snakes []*pb.Snake) (*pb.Point, error) {
+	openPoints, _ := getUnoccupiedPoints(width, height, food, snakes)
```

Please check the err return result here, or if an err is never returned, change the return result to not return an error.